src/DataTransformer/ValueObjectTransformer.php

```diff
@@ -59,7 +65,37 @@ public function transform($value)
 
     public function reverseTransform($value)
     {
-        return (new ViewDataInstantiator($this->form, $value))->instantiateObject();
+        try {
+            return (new ViewDataInstantiator($this->form, $value))->instantiateObject();
+        } catch (\Throwable $e) {
+            $exceptionHandlers = [];
+
+            if (null !== $this->form->getOption('handle_exception')) {
+                foreach ($this->form->getOption('handle_exception') as $exceptionClass) {
+                    $exceptionHandlers[] = new GenericExceptionHandler($exceptionClass);
+                }
+
+                $exceptionHandlers[] = $this->exceptionHandlerRegistry->get('type_error');
+            } else {
+                foreach ($this->form->getOption('exception_handling_strategy') as $strategy) {
+                    $exceptionHandlers[] = $this->exceptionHandlerRegistry->get($strategy);
+                }
+            }
+
+            if (1 === \count($exceptionHandlers)) {
+                $exceptionHandler = reset($exceptionHandlers);
+            } else {
+                $exceptionHandler = new ChainExceptionHandler($exceptionHandlers);
+            }
+
+            $error = $exceptionHandler->getError($this->form, $value, $e);
+
+            if (null !== $error) {
+                throw new TransformationFailedException($error->getMessage(), 0, $e);
```

Comment:
The fact that we extract the message from a `FormError` instance is a bit hacky. But I am a bit reluctant if creating a custom model for it is worth it. The `ArgumentTypeMismatchExceptionHandler`, for example, populates the "message template" and "parameters" properties of the `FormError` too which means that changing the `ExceptionHandlerInterface` so that `getError()` only returns a string doesn't look like an option.